include/nanobind/nb_traits.h

```diff
@@ -101,6 +101,20 @@ struct analyze_method<Ret (Cls::*)(Args...) const noexcept> {
     static constexpr size_t argc = sizeof...(Args);
 };
 
+template <typename F>
+struct strip_function_object {
+    using type = typename analyze_method<decltype(&F::operator())>::func;
+};
+
+// Extracts the function signature from a function, function pointer or lambda.
+template <typename Function, typename F = std::remove_reference_t<Function>>
+using function_signature_t = std::conditional_t<
```

Comment:
How does this relate to the other commit that is modifying function signatures? Are there simplification opportunities in merging them?